typescript/cli/src/commands/warp.ts

```diff
@@ -100,42 +106,86 @@ export const configure: CommandModuleWithContext<{
 };
 
 export const read: CommandModuleWithContext<{
-  chain: string;
-  address: string;
-  out: string;
+  chain?: string;
+  address?: string;
+  out?: string;
+  symbol?: string;
 }> = {
   command: 'read',
   describe: 'Reads the warp route config at the given path.',
   builder: {
+    symbol: {
+      type: 'string',
+      description: 'Identify warp route in registry by symbol',
+      demandOption: false,
+    },
     chain: {
       ...chainCommandOption,
-      demandOption: true,
+      demandOption: false,
     },
     address: addressCommandOption(
       'Address of the router contract to read.',
-      true,
+      false,
     ),
     out: outputFileCommandOption(),
   },
-  handler: async ({ context, chain, address, out }) => {
+  handler: async ({ context, chain, address, out, symbol }) => {
     logGray('Hyperlane Warp Reader');
     logGray('---------------------');
 
     const { multiProvider } = context;
-    const evmERC20WarpRouteReader = new EvmERC20WarpRouteReader(
-      multiProvider,
-      chain,
-    );
-    const warpRouteConfig = await evmERC20WarpRouteReader.deriveWarpRouteConfig(
-      address,
+
+    let addresses: ChainMap<string>;
+    if (symbol) {
+      const matching = await context.registry.getWarpRoutes({
+        symbol,
+      });
+      const routes = Object.entries(matching);
+
+      let warpCoreConfig: WarpCoreConfig;
+      if (routes.length === 0) {
+        logRed(`No warp routes found for symbol ${symbol}`);
+        process.exit(0);
+      } else if (routes.length === 1) {
+        warpCoreConfig = routes[0][1];
+      } else {
+        logGreen(`Multiple warp routes found for symbol ${symbol}`);
+        const chosenRouteId = await select({
+          message: 'Select from matching warp routes',
+          choices: routes.map(([routeId, _]) => ({
+            value: routeId,
+          })),
+        });
+        warpCoreConfig = matching[chosenRouteId];
+      }
+
+      addresses = Object.fromEntries(
+        warpCoreConfig.tokens.map((t) => [t.chainName, t.addressOrDenom!]),
+      );
+    } else if (chain && address) {
+      addresses = {
+        [chain]: address,
+      };
+    } else {
+      logGreen(`Please specify either a symbol or chain and address`);
+      process.exit(0);
+    }
```

Comment:
this seems to overlap with
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/b0828b3d0eb95921a8753bb0c08b36b37ca0af15/typescript/cli/src/send/transfer.ts#L119-L130

can we consolidate somehow?

Reply:
cc @jmrossy since we discussed this in-depth last week

Reply:
I will add the same symbol option to this command though